ros2_batch_job/util.py

```diff
@@ -169,10 +169,26 @@ class MyProtocol(AsyncSubprocessProtocol):
     def __init__(self, cmd, exit_on_error, *args, **kwargs):
         self.cmd = cmd
         self.exit_on_error = exit_on_error
+        self.progress_bar = False
+        self.progress_data = []
         AsyncSubprocessProtocol.__init__(self, *args, **kwargs)
 
     def on_stdout_received(self, data):
-        sys.stdout.write(data.decode('utf-8', 'replace').replace(os.linesep, '\n'))
+        # '[?25l' and '[?25h' can be considered as a sign of begining and end of pip progress bar
```

Comment:
I'm looking for something describing why we are doing this (though knowing the mechanism a bit can be helpful).  Something more like:

```suggestion
        # The 'pip install' progress bar output can be pretty noisy in CI output.  Print only every 10th
        # output to reduce the noisiness.  We detect the pip progress bar by the beginning sequence of
        # '[?25l' and the end sequence of '[?25h'.
```

Reply:
Much thanks for this better explanation!